pageserver/src/tenant/blob_io.rs

```diff
@@ -238,14 +246,16 @@ where
         cancel: CancellationToken,
         ctx: &RequestContext,
         flush_task_span: tracing::Span,
-    ) -> anyhow::Result<Self> {
+    ) -> Result<Self, BlobWriterError> {
+        let gate_token = gate.enter().map_err(|e| BlobWriterError::Other(e.into()))?;
```

Comment:
entering the gate should map to `Cancelled`; fix that not just here but all over this PR